Elasticsearch/Document/Source.php

```diff
@@ -21,35 +22,28 @@ final class EMSSource implements EMSSourceInterface
     /** @var array */
     private $source;
 
-    public const FIELD_CONTENT_TYPE = '_contenttype';
-    public const FIELD_FINALIZED_BY = '_finalized_by';
-    public const FIELD_FINALIZATION_DATETIME = '_finalization_datetime';
-    public const FIELD_HASH = '_sha1';
-    public const FIELD_PUBLICATION_DATETIME = '_published_datetime';
-
-
     public function __construct(array $source)
     {
-        $this->contentType = $source[self::FIELD_CONTENT_TYPE] ?? null;
-        $this->finalizedBy = $source[self::FIELD_FINALIZED_BY] ?? null;
-        $this->hash = $source[self::FIELD_HASH] ?? null;
+        $this->contentType = $source['_contenttype'] ?? null;
+        $this->finalizedBy = $source['_finalized_by'] ?? null;
+        $this->hash = $source['_sha1'] ?? null;
```

Comment:
Unrelated to your change, but doesn't phpstan requires that the type hint of these properties also can be null ? (starting line 13)

Reply:
Mmm okay that's true, something to think about. Because ems sources should (must) always have these, but we may also need a normal source object. Maybe emsSource was not so bad after all ...

Reply:
I just don't understand why phpstan is not complaining, as it is supposed to be on level 7!

Reply:
But what if someone creates a contenttype with name "unknown"? This might cause issues. This would mean that we create a reserved keyword in our system.